staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -58,6 +58,9 @@ func (s *SecureServingInfo) Serve(handler http.Handler, shutdownTimeout time.Dur
 			// enable HTTP2 for go's 1.7 HTTP Server
 			NextProtos: []string{"h2", "http/1.1"},
 		},
+		// TODO: make these configurable
+		ReadTimeout: 10 * time.Second,
+		IdleTimeout: 5 * time.Minute,
```

Comment:
Same question about watch connections

Reply:
No

Reply:
these are the API requests we need to make sure are not broken by this:
* long watch stream (both http chunking and websocket forms)
* log-following streaming (both http streaming and websocket forms)
* exec/attach (both spdy and websocket forms)
* portforward

we need to ensure that this idle timeout does not prematurely close these connections. I know some of those hijack connections, but I didn't check that all of them do.

can you set IdleTimeout short (e.g. 10 seconds), and make sure that all of those API calls (in both forms) are able to be held open (even if there are no watch events or exec frames sent during that time period, etc)

Reply:
@liggitt Yes, I have tested this both with kubectl and curl, both work well without connection closed in 10s even longer.